include/numerics/dense_matrix.h

```diff
@@ -34,6 +34,21 @@
 #include <vector>
 #include <algorithm>
 
+#ifdef LIBMESH_HAVE_METAPHYSICL
+namespace MetaPhysicL
+{
+template <typename, typename>
+class DualNumber;
+}
+namespace std
+{
+template <typename T, typename D>
```

Comment:
I guess what you are doing here is making it so that `std::abs(x)`, where `x` is a `DualNumber` compiles? Would you mind adding a comment to that effect?

Reply:
👍

Reply:
comments added

Reply:
Interesting, we have a bunch of forward declarations like `template <typename T> class DenseVector;`, it might make sense to go delete the `T` parameter from all of them as it doesn't really add much.